bound/typescript/src/presentation-definition.ts

```diff
@@ -0,0 +1,56 @@
+export type PresentationDefinition = {
+  id: string;
+  input_descriptors: InputDescriptor[];
+  name?: string;
+  purpose?: string;
+  submission_requirements?: SubmissionRequirement[];
+};
+
+export type InputDescriptor = {
+  constraints: Constraints;
+  id: string;
+  name?: string;
+  purpose?: string;
+};
+
+export type Constraints = {
+  fields: Field[];
+};
+
+export type Field = {
+  filter?: Filter;
+  id?: string;
+  name?: string;
+  optional?: boolean;
+  path: string[];
+  predicate?: Optionality;
+  purpose?: string;
+};
+
+export type Filter = {
+  const?: string;
+  contains?: Filter;
+  pattern?: string;
+  type?: string;
+};
+
+// todo make enum
+export type Optionality = {
+  optionality: string;
+};
+
+export type SubmissionRequirement = {
+  count?: number;
+  from?: string;
+  from_nested?: SubmissionRequirement[];
+  max?: number;
+  min?: number;
+  name?: string;
+  purpose?: string;
+  rule: SubmissionRequirementRule;
+};
+
+// todo make enum
```

Comment:
I would want to make this and `Optionality` above actual ts enums, or type unions to be more exact